pkg/controller/plan/kubevirt.go

```diff
@@ -1835,6 +1832,103 @@ func (r *KubeVirt) EnsurePersistentVolume(vmRef ref.Ref, persistentVolumes []cor
 	return
 }
 
+func (r *KubeVirt) CreatePvcForNfs(pvcName string) (err error) {
+	//TODO: set ownerReferenceso the PV and PVC deleted once the plan is done.
+	// ownerReference := meta.OwnerReference{
+	// 	APIVersion: "forklift.konveyor.io/v1beta1",
+	// 	Kind:       "Plan",
+	// 	Name:       r.Plan.Name,
+	// 	UID:        r.Plan.UID,
+	// }
+
+	sourceProvider := r.Source.Provider
+	pvName := fmt.Sprintf("ova-server-pv-%s", r.Plan.Name)
+	splitted := strings.Split(sourceProvider.Spec.URL, ":")
+
+	if len(splitted) != 2 {
+		r.Log.Error(nil, "NFS server path doesn't contains :")
+	}
+	nfsServer := splitted[0]
+	nfsPath := splitted[1]
+
+	pv := &v1.PersistentVolume{
+		ObjectMeta: meta.ObjectMeta{
+			Name: pvName,
+		},
+		Spec: v1.PersistentVolumeSpec{
+			Capacity: v1.ResourceList{
+				v1.ResourceStorage: resource.MustParse("1Gi"),
+			},
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadOnlyMany,
+			},
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				NFS: &v1.NFSVolumeSource{
+					Path:   nfsPath,
+					Server: nfsServer,
+				},
+			},
+		},
+	}
+	err = r.Create(context.TODO(), pv)
+	if err != nil {
+		r.Log.Error(err, "Failed to create OVA plan PV")
+		return
+	}
+
+	sc := ""
+	pvc := &v1.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      pvcName,
+			Namespace: r.Plan.Spec.TargetNamespace,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceStorage: resource.MustParse("1Gi"),
+				},
+			},
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				v1.ReadOnlyMany,
+			},
+			VolumeName:       pvName,
+			StorageClassName: &sc,
+		},
+	}
+	err = r.Create(context.TODO(), pvc)
+	if err != nil {
+		r.Log.Error(err, "Failed to create OVA plan PVC")
+		return
+	}
+
+	// wait until pvc and pv are bounded.
+	timeout := time.After(5 * time.Minute)
+	tick := time.Tick(5 * time.Second)
+	pvcNamespacedName := types.NamespacedName{
+		Namespace: r.Plan.Spec.TargetNamespace,
+		Name:      pvcName,
+	}
+
+	for {
+		select {
+		case <-timeout:
+			r.Log.Error(err, "Timed out waiting for PVC to be bound")
+			return
```

Comment:
will it return an error or we will reconcile and try again? if we will try again - will we have the PV or PVC existing and try to create the same ones?